### DISCUSSION TITLE
Build not working

### DISCUSSION THREAD

**Discussion by mats68** [vercel/next.js#18186] [Help] [answered]
Hello 
I tried next for the first time. 
I created a new app with typescript. I added some pages and styles
everything worked in dev-mode as expected.

Unfortunately when I try to build SSG with "next build" and "next export"
no styles are there and the links are not working.

**Comment by jamesmosier** [comment, marked as answer]
Hi @mats68. Do you have an example repository you could share?

**Reply by mats68** [reply]
Hi @jamesmosier 

Yes sure its on https://github.com/mats68/next-text-ssg

**Reply by jamesmosier** [reply]
I was able to run it successfully using `next build && next export` and then running `serve out`.

If you try to just open up the `out/index.html` file it won't work, you need to serve the assets from a server.

[serve](https://www.npmjs.com/package/serve) is a great solution that allows you to "serve" any directory on a port.

**Reply by mats68** [reply]
@jamesmosier Thanks a lot man, that works great ! 

Is it also possible to configure that just opening `out/index.html` also works ?

**Reply by jamesmosier** [reply]
Ultimately it is a security feature of browsers. In order to request CSS/JS files from an HTML document it needs to know where to find them. That is what the local server using `serve` does - it allows your HTML document to know where to find those other resources.
